.travis.yml

```diff
@@ -19,17 +19,24 @@ env:
         # Make sure beaver is in the PATH
         - PATH="$(git config -f .gitmodules submodule.beaver.path)/bin:$PATH"
     matrix:
-        - DC=dmd1 DIST=trusty F=production
-        - DC=dmd1 DIST=trusty F=devel
-        - DC=dmd1 DIST=xenial F=production AFTER_SCRIPT=1
-        - DC=dmd1 DIST=xenial F=devel
-        - DC=dmd-transitional DIST=trusty F=production
-        - DC=dmd-transitional DIST=trusty F=devel
-        - DC=dmd-transitional DIST=xenial F=production
-        - DC=dmd-transitional DIST=xenial F=devel
+        - DMD=1.080.0 DIST=trusty F=production
+        - DMD=1.080.0 DIST=trusty F=devel
+        - DMD=1.080.0 DIST=xenial F=production AFTER_SCRIPT=1
+        - DMD=1.080.0 DIST=xenial F=devel
+        - DMD=2.070.2.s12 DIST=trusty F=production
+        - DMD=2.070.2.s12 DIST=trusty F=devel
+        - DMD=2.070.2.s12 DIST=xenial F=production
+        - DMD=2.070.2.s12 DIST=xenial F=devel
 
-install: beaver install
+install:
+    - beaver docker build --build-arg "DMD=$DMD" --build-arg "DC=$(ci/DC.sh)" -f "Dockerfile${DIST:+.$DIST}" .
 
-script: BEAVER_DOCKER_VARS="DC" beaver run ci/travis.sh
+script:
+    - beaver run ci/test.sh
 
-after_success: ci/travis-after-success.sh
+after_success:
+    # run only once among matrix rows, workaround for Travis lack of support
+    # for sequential pipelines
+    - test "$AFTER_SCRIPT" || exit 0
+    - beaver run ci/codecov.sh
```

Comment:
The problem with this is it will share the credentials with codecov (`beaver run` shares Bintray credential with the docker container, this should probably done only with `beaver bintray ...`, but still there might be other credential sharing in the future with `beaver run`), which is what we wanted to avoid in the first place by running it inside a docker container. Ideally we should move the codecov magic to beaver so we don't have to care about these details in the projects, but for now we should call docker differently to run codecov.